pkg/app/piped/analysisprovider/metrics/datadog/datadog.go

```diff
@@ -75,6 +68,9 @@ func NewProvider(apiKey, applicationKey string, queriedTimePeriod time.Duration,
 type Option func(*Provider)
 
 func WithAddress(address string) Option {
+	if address == "" {
+		address = defaultAddress
+	}
```

Comment:
nits, this one looks redundant to me 👀 Should we remove it just like we do in `WithTimeout`?

Reply:
Absolutely. I'm going to undo it right now.